src/modules/previewpane/MarkdownPreviewHandler/MarkdownPreviewHandlerControl.cs

```diff
@@ -0,0 +1,76 @@
+
+        /// <summary>
+        /// Markdown HTML footer.
+        /// </summary>
+        private readonly string htmlFooter = "</div></body></html>";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MarkdownPreviewHandlerControl"/> class.
+        /// </summary>
+        public MarkdownPreviewHandlerControl()
+        {
+            this.extension = new HTMLParsingExtension();
+            this.pipelineBuilder = new MarkdownPipelineBuilder().UseAdvancedExtensions().UseEmojiAndSmiley();
+            this.pipelineBuilder.Extensions.Add(this.extension);
+        }
+
+        /// <summary>
+        /// Start the preview on the Control.
+        /// </summary>
+        /// <param name="dataSource">Path to the file.</param>
+        public override void DoPreview<T>(T dataSource)
+        {
+            this.InvokeOnControlThread(() =>
+            {
+                string filePath = dataSource as string;
+                string fileText = File.ReadAllText(filePath);
+                this.extension.BaseUrl = Path.GetDirectoryName(filePath);
+
+                MarkdownPipeline pipeline = this.pipelineBuilder.Build();
+                string parsedMarkdown = Markdown.ToHtml(fileText, pipeline);
+                string html = this.htmlHeader + parsedMarkdown + this.htmlFooter;
+
+                // File.WriteAllText("C:\\Users\\divyan\\Desktop\\output.html", html);

```

Comment:
[NIT] I guess this comment can be removed?

Reply:
Thanks for pointing out! I will update this